### ISSUE TITLE
Regression in dynamodb type hints

### ISSUE THREAD

**Issue by wiget** [youtype/mypy_boto3_builder#166]
**Describe the bug**
Version 1.24.55.post1 of mypy-boto3-dynamodb introduces regression for, at least, `GetItemOutputTypeDef`

Code that previously does not reported any type errors started reporting errors. The last version that worked is 1.24.55

**To Reproduce**

1. Install `boto3-stubs[dynamodb]`
2. Run `mypy`/`pyright` on the following code sample

```python
import boto3
from mypy_boto3_dynamodb import DynamoDBClient

client: DynamoDBClient = boto3.client("dynamodb")
response = client.get_item(
        TableName="dummy-tabele",
        Key={"Key": {"S": "key1"}}
        )
value = response["Item"]["property"]
print(value["S"])
```

**Actual output**

```
dynamodb.py:10: error: No overload variant of "__getitem__" of "bytes" matches argument type "str"
dynamodb.py:10: note: Possible overload variants:
dynamodb.py:10: note:     def __getitem__(self, SupportsIndex) -> int
dynamodb.py:10: note:     def __getitem__(self, slice) -> bytes
dynamodb.py:10: error: No overload variant of "__getitem__" of "bytearray" matches argument type "str"
dynamodb.py:10: note:     def __getitem__(self, slice) -> bytearray
dynamodb.py:10: error: Value of type "Union[AttributeValueTypeDef, bytes, bytearray, str, int, Decimal, bool, Set[int], Set[Decimal], Set[str], Set[bytes], Set[bytearray], Sequence[Any], Mapping[str, Any], None]" is not indexable
dynamodb.py:10: error: No overload variant of "__getitem__" of "Sequence" matches argument type "str"
dynamodb.py:10: note:     def __getitem__(self, int) -> Any
dynamodb.py:10: note:     def __getitem__(self, slice) -> Sequence[Any]
dynamodb.py:10: error: Invalid index type "str" for "Union[AttributeValueTypeDef, bytes, bytearray, str, int, Decimal, bool, Set[int], Set[Decimal], Set[str], Set[bytes], Set[bytearray], Sequence[Any], Mapping[str, Any], None]"; expected type "Union[SupportsIndex, slice]"
Found 5 errors in 1 file (checked 1 source file)
```

**Expected output**

```
```

**Additional context**
Tested on Ubuntu 22.04, python 3.10.4.

 `boto3-stubs` installed using `poetry add "boto3-stubs[dynamodb]"`.

Packages installed in virtual environment used for testing:
```
❯ poetry show
boto3               1.24.59       The AWS SDK for Python
boto3-stubs         1.24.59       Type annotations for boto3 1.24.59 generated with mypy-boto3-builder 7.11.7
botocore            1.27.59       Low-level, data-driven core of boto 3.
botocore-stubs      1.27.59       Type annotations and code completion for botocore
jmespath            1.0.1         JSON Matching Expressions
mypy                0.971         Optional static typing for Python
mypy-boto3-dynamodb 1.24.55.post1 Type annotations for boto3.DynamoDB 1.24.55 service generated with mypy-boto3-builder 7.11.7
mypy-extensions     0.4.3         Experimental type system extensions for programs checked with the mypy typechecker.
python-dateutil     2.8.2         Extensions to the standard Python datetime module
s3transfer          0.6.0         An Amazon S3 Transfer Manager
six                 1.16.0        Python 2 and 3 compatibility utilities
tomli               2.0.1         A lil' TOML parser
types-awscrt        0.14.0        Type annotations and code completion for awscrt
types-s3transfer    0.6.0.post4   Type annotations and code completion for s3transfer
typing-extensions   4.3.0         Backported and Experimental Type Hints for Python 3.7+
urllib3             1.26.12       HTTP library with thread-safe connection pooling, file post, and more.
```

Most probably [this](https://github.com/youtype/mypy_boto3_builder/commit/873c104eac16c05e3149a6b90c228cb9aada0b4f) commit introduces regression.

**Comment by vemel** [comment]
Thank you, will check. Yes, it has been changed it one of the latest releases to accept both `botocore` and `boto3`-style `AttributeValue`.

**Comment by vemel** [comment]
Fixed in `mypy-boto3-dynamodb 1.24.60`.

Please check and let me know if it works as it should.

**Comment by wiget** [comment]
Now it works as expected. Thanks.
